src/strtree.c

```diff
@@ -52,8 +52,16 @@ static PyArrayObject *copy_kvec_to_npy(npy_intp_vec *arr)
 static void STRtree_dealloc(STRtreeObject *self)
 {
     void *context = geos_context[0];
+    size_t i, size;
+    // free the tree
     if (self->ptr != NULL) { GEOSSTRtree_destroy_r(context, self->ptr); }
-    Py_XDECREF(self->geometries);
+    // free the geometries
+    size = kv_size(self->_geoms);
+    for (i = 0; i < size; i++) {
+        Py_XDECREF(kv_pop(self->_geoms));
```

Comment:
@caspervdw 
I think this can be `Py_XDECREF(kv_A(self->_geoms, i));`, which avoids decrementing the size of the vector (one less operation).

I'll add this in #108